trajopt/include/trajopt/kinematic_terms.hpp

```diff
@@ -29,18 +29,12 @@ struct CartPoseErrCalculator : public VectorOfVector
   {
   }
 
-  VectorXd operator()(const VectorXd& dof_vals) const;
-};
+  void Plot(const tesseract::BasicPlottingPtr& plotter, const VectorXd& dof_vals) override;
 
-struct CartPoseErrorPlotter : public Plotter
-{
-  std::shared_ptr<void> m_calc;  // actually points to a CartPoseErrCalculator = CartPoseCost::f_
-  VarVector m_vars;
-  CartPoseErrorPlotter(std::shared_ptr<void> calc, const VarVector& vars) : m_calc(calc), m_vars(vars) {}
-  void Plot(const tesseract::BasicPlottingPtr plotter, const DblVec& x);
+  VectorXd operator()(const VectorXd& dof_vals) const;
 };
 
-struct StaticCartPoseErrCalculator : public VectorOfVector
+struct StaticCartPoseErrCalculator : public TrajOptVectorOfVector
```

Comment:
While I understand you are debugging, I am a little concerned that you've opted everyone into paying this cost for plotting on what is probably the most common of error functions.

Furthermore `TrajOptVectorOfVector` is not a descriptive name: perhaps something that indicates the interfaces you've added? E.g. `PlottableVectorOfVector`.

Reply:
Are you speaking specifically about this cost function? 

Those that implement there own cost or constraint can still use the VectorOfVector type.